source/common/stats/tag_extractor_impl.h

```diff
@@ -50,8 +49,31 @@ class TagExtractorImpl : public TagExtractor {
   const std::string name_;
   const std::string prefix_;
   const std::string substr_;
+};
+
+class TagExtractorStdRegexImpl : public TagExtractorImpl {
```

Comment:
TagExtractorImpl is an abstract base class, consider renaming it to TagExtractorImplBase.